synapse/http/federation/well_known_resolver.py

```diff
@@ -111,18 +129,28 @@ def get_well_known(self, server_name):
 
             result = None
 
-            # add some randomness to the TTL to avoid a stampeding herd every hour
-            # after startup
-            cache_period = WELL_KNOWN_INVALID_CACHE_PERIOD
-            cache_period += random.uniform(0, WELL_KNOWN_DEFAULT_CACHE_PERIOD_JITTER)
+            if self._had_valid_well_known_cache.get(server_name, False):
+                # We have recently seen a valid well-known record for this
+                # server, so we cache the lack of well-known for a shorter time.
+                cache_period = WELL_KNOWN_DOWN_CACHE_PERIOD
+                cache_period += random.uniform(
+                    0, WELL_KNOWN_DEFAULT_CACHE_PERIOD_JITTER
```

Comment:
the jitter is bigger than the base value here... are you sure?

Reply:
Oh, yeah. I'm tempted to change the jitter to be a factor, e.g. 10%, thoughts?